### PR TITLE
Update tests to support start year of 0001

### PR DESCRIPTION
**Pull request by matthewhoffman**
MALI's Registry has been using a default `config_start_time` of '0000-01-01_00:00:00', which is not CF-compliant.  The MALI-Dev PR https://github.com/MALI-Dev/E3SM/pull/85 updates Registry to make the default `config_start_time` '0001-01-01_00:00:00'.  This PR updates the few compass tests that were assuming a start year of 0000.

Checklist
* [x] The `MALI-Dev` submodule has been updated with relevant MALI changes
* [x] Document (in a comment titled `Testing` in this PR) any testing that was used to verify the changes

### CODE REVIEW

**Comment by matthewhoffman** [comment]
# Testing

Prior to this PR and using MALI version in  https://github.com/MALI-Dev/E3SM/pull/85, I get these results:
```
Test Runtimes:
00:15 PASS landice_dome_2000m_sia_restart_test
00:52 PASS landice_dome_2000m_sia_decomposition_test
00:10 PASS landice_dome_variable_resolution_sia_restart_test
00:18 PASS landice_dome_variable_resolution_sia_decomposition_test
00:14 FAIL landice_enthalpy_benchmark_A
00:29 PASS landice_eismint2_decomposition_test
00:19 PASS landice_eismint2_enthalpy_decomposition_test
00:21 PASS landice_eismint2_restart_test
00:28 PASS landice_eismint2_enthalpy_restart_test
00:19 PASS landice_greenland_sia_restart_test
00:14 PASS landice_greenland_sia_decomposition_test
00:07 FAIL landice_hydro_radial_restart_test
00:11 FAIL landice_hydro_radial_decomposition_test
00:19 PASS landice_dome_2000m_fo_decomposition_test
00:27 PASS landice_dome_2000m_fo_restart_test
00:13 PASS landice_dome_variable_resolution_fo_decomposition_test
00:18 PASS landice_dome_variable_resolution_fo_restart_test
00:21 PASS landice_circular_shelf_decomposition_test
00:51 PASS landice_greenland_fo_decomposition_test
00:47 PASS landice_greenland_fo_restart_test
00:37 PASS landice_thwaites_fo_decomposition_test
00:40 PASS landice_thwaites_fo_restart_test
00:13 PASS landice_thwaites_fo-depthInt_decomposition_test
00:21 PASS landice_thwaites_fo-depthInt_restart_test
00:51 PASS landice_humboldt_mesh-3km_restart_test_velo-fo_calving-von_mises_stress_damage-threshold_faceMelting
00:23 PASS landice_humboldt_mesh-3km_restart_test_velo-fo-depthInt_calving-von_mises_stress_damage-threshold_faceMelting
Total runtime 10:44
FAIL: 3 tests failed, see above.
```

With this branch and the same version of MALI, the failing tests now pass:
```
Test Runtimes:
00:16 PASS landice_dome_2000m_sia_restart_test
00:05 PASS landice_dome_2000m_sia_decomposition_test
00:07 PASS landice_dome_variable_resolution_sia_restart_test
00:05 PASS landice_dome_variable_resolution_sia_decomposition_test
00:21 PASS landice_enthalpy_benchmark_A
00:13 PASS landice_eismint2_decomposition_test
00:17 PASS landice_eismint2_enthalpy_decomposition_test
00:18 PASS landice_eismint2_restart_test
00:14 PASS landice_eismint2_enthalpy_restart_test
00:16 PASS landice_greenland_sia_restart_test
00:12 PASS landice_greenland_sia_decomposition_test
00:12 PASS landice_hydro_radial_restart_test
00:11 PASS landice_hydro_radial_decomposition_test
00:27 FAIL landice_humboldt_mesh-3km_decomposition_test_velo-none_calving-none_subglacialhydro
00:59 FAIL landice_humboldt_mesh-3km_restart_test_velo-none_calving-none_subglacialhydro
00:30 PASS landice_dome_2000m_fo_decomposition_test
00:20 PASS landice_dome_2000m_fo_restart_test
00:25 PASS landice_dome_variable_resolution_fo_decomposition_test
00:41 PASS landice_dome_variable_resolution_fo_restart_test
01:03 PASS landice_circular_shelf_decomposition_test
01:12 PASS landice_greenland_fo_decomposition_test
01:20 PASS landice_greenland_fo_restart_test
00:44 PASS landice_thwaites_fo_decomposition_test
01:04 PASS landice_thwaites_fo_restart_test
00:32 PASS landice_thwaites_fo-depthInt_decomposition_test
00:33 PASS landice_thwaites_fo-depthInt_restart_test
00:42 PASS landice_humboldt_mesh-3km_restart_test_velo-fo_calving-von_mises_stress_damage-threshold_faceMelting
00:19 PASS landice_humboldt_mesh-3km_restart_test_velo-fo-depthInt_calving-von_mises_stress_damage-threshold_faceMelting
Total runtime 13:43
FAIL: 2 tests failed, see above.
```
Note that my baseline was using an older version of compass that did not include the two Humboldt subglacial hydro tests that now fail.  Both of those tests pass validation and only report failure because the baselines do not exist in my test.  Both use a start time of `2007-07-01` and will not be impacted by the change being tested here.  Thus, this testing demonstrates that this compass PR fixes the tests affected by https://github.com/MALI-Dev/E3SM/pull/85.

File: compass/landice/tests/enthalpy_benchmark/A/namelist.landice2 [hunk 1]
```diff
@@ -1,5 +1,5 @@
 config_surface_air_temperature_value = 268.15
 config_do_restart = .true.
-config_start_time = '100000-01-01_00:00:00'
+config_start_time = '100001-01-01_00:00:00'
```
Comment (trhille) [inline compass/landice/tests/enthalpy_benchmark/A/namelist.landice2:3]: Is this change necessary?

File: compass/landice/tests/enthalpy_benchmark/A/namelist.landice3 [hunk 2]
```diff
@@ -1,4 +1,4 @@
 config_do_restart = .true.
-config_start_time = '150000-01-01_00:00:00'
+config_start_time = '150001-01-01_00:00:00'
```
Comment (trhille) [inline compass/landice/tests/enthalpy_benchmark/A/namelist.landice3:2]: Is this change necessary?

**Review by trhille — COMMENTED** [review]
@matthewhoffman, thanks for taking care of this. There are a few start times that have been updated that may not need to be.

File: compass/landice/tests/enthalpy_benchmark/A/namelist.landice2 (hunk 1 above)
Comment (matthewhoffman) [reply compass/landice/tests/enthalpy_benchmark/A/namelist.landice2:3]: Yes - this is a restart after 10,000 years of the first phase.  Now that the first phase start time has moved from year 0 to 1, this restart start time needs to be advanced one year as well.

File: compass/landice/tests/enthalpy_benchmark/A/namelist.landice3 (hunk 2 above)
Comment (matthewhoffman) [reply compass/landice/tests/enthalpy_benchmark/A/namelist.landice3:2]: This is phase 3, so same answer as previous.

**Review by trhille — APPROVED** [review]
Approved based on inspection and @matthewhoffman's testing.
